ichspi.c

```diff
@@ -1950,13 +1961,18 @@ static enum ich_access_protection ich9_handle_region_access(struct fd_region *fd
 #define ICH_PR_PERMS(pr)	(((~((pr) >> PR_RP_OFF) & 1) << 0) | \
 				 ((~((pr) >> PR_WP_OFF) & 1) << 1))
 
-static enum ich_access_protection ich9_handle_pr(const size_t reg_pr0, unsigned int i)
+static enum ich_access_protection ich9_handle_pr(const size_t reg_pr0, unsigned int i, struct protected_range* prot)
 {
 	uint8_t off = reg_pr0 + (i * 4);
 	uint32_t pr = mmio_readl(ich_spibar + off);
 	unsigned int rwperms_idx = ICH_PR_PERMS(pr);
 	enum ich_access_protection rwperms = access_perms_to_protection[rwperms_idx];
 
+	prot->base = ICH_FREG_BASE(pr);
+	prot->limit = ICH_FREG_LIMIT(pr);
+	prot->write_prot = rwperms == WRITE_PROT;
```

Comment:
```suggestion
	prot->write_prot = (rwperms == WRITE_PROT) || (rwperms == LOCKED);
```

I think, at least

Reply:
I assumed based on https://github.com/Dasharo/flashrom/blob/ababbe13e6b06be551cc0f28a538b4747fcd2115/ichspi.c#L2267 that LOCKED != write-protected
If I'm reading this correctly in case `ich_spi_rw_restricted == LOCKED` then warning will only mention that's it's read-protected